src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java

```diff
@@ -251,53 +270,39 @@ static OracleReactiveJdbcAdapter getInstance() {
    * {@link OracleDataSource} that implements the Reactive Extensions APIs for
    * creating connections.
    * </p>
-   *
-   * <h3>Required Standard Options</h3>
+   * <h3>Composing a JDBC URL</h3>
    * <p>
-   * This implementation requires values to be set for the following options:
-   * </p><ul>
-   *   <li>{@link ConnectionFactoryOptions#HOST}</li>
-   * </ul><p>
-   * The values set for these options are used to compose an Oracle JDBC URL as:
+   * The {@code options} provided to this method are used to compose a URL
+   * for the JDBC {@code DataSource}. Values for standard
+   * {@link ConnectionFactoryOptions} of {@code HOST}, {@code PORT}, and
+   * {@code DATABASE} are used to compose the JDBC URL with {@code DATABASE}
+   * interpreted as a service name (not a system identifier (SID)):
    * </p><pre>
-   * jdbc:oracle:thin:@HOST
+   *   jdbc:oracle:thin:@HOST:PORT/DATABASE
    * </pre><p>
-   * This minimal JDBC URL may specify a TNS alias as a the HOST value when
-   * Oracle JDBC is configured to read a tnsnames.ora file.
-   * </p>
-   *
-   * <h3>Optional Standard Options</h3>
-   * <p>
-   * This implementation supports optional values that are set for the
-   * following options:
-   * </p><ul>
-   *   <li>{@link ConnectionFactoryOptions#PORT}</li>
-   *   <li>{@link ConnectionFactoryOptions#DATABASE}</li>
-   *   <li>{@link ConnectionFactoryOptions#USER}</li>
-   *   <li>{@link ConnectionFactoryOptions#PASSWORD}</li>
-   *   <li>{@link ConnectionFactoryOptions#CONNECT_TIMEOUT}</li>
-   *   <li>{@link ConnectionFactoryOptions#SSL}</li>
-   * </ul><p>
-   * When PORT and DATABASE are present, an Oracle JDBC URL is composed as:
+   * Alternatively, the host, port, and service name may be specified using an
+   * <a href="https://docs.oracle.com/en/database/oracle/oracle-database/21/netag/identifying-and-accessing-database.html#GUID-8D28E91B-CB72-4DC8-AEFC-F5D583626CF6"></a>
+   * Oracle Net Descriptor</a>. The descriptor may be set as the value of an
+   * {@link Option} having the name "descriptor". When the descriptor option is
+   * present, the JDBC URL is composed as:
    * </p><pre>
-   * jdbc:oracle:thin:@HOST:PORT/DATABASE
+   *   jdbc:oracle:thin:@(DESCRIPTION=...)
    * </pre><p>
-   * Note that {@code DATABASE} is interpreted as the service name of an Oracle
-   * Database; It is not interpreted as a system identifier (SID).
-   * </p><p>
-   * Values set for {@code USER} and {@code PASSWORD} options are used to
-   * authenticate with an Oracle Database.
+   * When the "descriptor" option is provided, it is invalid to specify any
+   * other options that might conflict with values also specified in the
+   * descriptor. For instance, the descriptor element of
+   * {@code (ADDRESSS=(HOST=...)(PORT=...)(PROTOCOL=...))} specifies values
+   * that overlap with the standard {@code Option}s of {@code HOST}, {@code
+   * PORT}, and {@code SSL}. An {@code IllegalStateException} is thrown
```

Comment:
Would IllegalArgumentException make more sense? Just thinking out loud.

Reply:
Yes, I think you are right. Reviewing the spec, this case seems to meet the criteria for IllegalArugmentException:

> Drivers throw IllegalArgumentException if a method has been received an illegal or inappropriate argument (such as values that are out of bounds or an expected parameter is null). This exception is a generic exception that is not associated with an error code or an SQLState

https://r2dbc.io/spec/0.8.4.RELEASE/spec/html/#exceptions

Reply:
You still have IllegalStateException mentioned here.

Reply:
@Michael-A-McMahon you still have IllegalStateException mentioned in the javadoc (line 296). This needs to be changed to IllegalArugmentException.

Reply:
Thanks for catching that, it's now fixed in the latest commit